app/controllers/spree/api/v2/storefront/checkout_controller_decorator.rb

```diff
@@ -46,6 +46,19 @@ def payment_redirect
             end
           end
 
+          def request_update_all_payments
+            order = spree_current_order
+
+            return render_serialized_payload { serialize_resource(order) } if order.paid?
+
```

Comment:
Add a comment on top of it, explaining the usecase why we need to loop through the payments